packages/sync-actions/src/order-actions.js

```diff
@@ -146,42 +146,36 @@ export function actionsMapReturnsInfo(diff, oldObj, newObj) {
       }
       return []
     },
-    [CHANGE_ACTIONS]: (oldSReturnInfo, newReturnInfo) => {
-      const updateActions = Object.keys(returnInfoDiff).reduce(
-        (itemActions, key) => {
-          const { items = {} } = returnInfoDiff[key]
-          if (Object.keys(items).length > 0) {
-            return [
-              ...itemActions,
-              ...Object.keys(items).reduce((actions, index) => {
-                const itActions = []
-                const item = newReturnInfo.items[index]
-                if (items[index].shipmentState) {
-                  itActions.push({
-                    action: 'setReturnShipmentState',
-                    returnItemId: item.id,
-                    shipmentState: item.shipmentState,
-                  })
-                }
-                if (items[index].paymentState) {
-                  itActions.push({
-                    action: 'setReturnPaymentState',
-                    returnItemId: item.id,
-                    paymentState: item.paymentState,
-                  })
-                }
-
-                return [...actions, ...itActions]
-              }, []),
-            ]
-          }
-
-          return itemActions
-        },
-        []
-      )
+    [CHANGE_ACTIONS]: (oldSReturnInfo, newReturnInfo, key) => {
+      const { items = {} } = returnInfoDiff[key]
+      const itemActions = []
+      if (Object.keys(items).length > 0) {
+        return [
+          ...itemActions,
+          ...Object.keys(items).reduce((actions, index) => {
+            const itActions = []
+            const item = newReturnInfo.items[index]
+            if (items[index].shipmentState) {
+              itActions.push({
+                action: 'setReturnShipmentState',
+                returnItemId: item.id,
+                shipmentState: item.shipmentState,
+              })
+            }
+            if (items[index].paymentState) {
+              itActions.push({
+                action: 'setReturnPaymentState',
+                returnItemId: item.id,
+                paymentState: item.paymentState,
+              })
+            }
+
+            return [...actions, ...itActions]
+          }, []),
+        ]
```

Comment:
Definitely worth fixing as long as we're modifying the code here. It might be wise to take some time to grep the entire codebase for `.reduce` and knock them all down.

Reply:
In this case, and because this is public library we have to be very careful about the context of the changes.

This would be a patch release, which means that it goes with X.X.X to X.X.X+1.

In the scenario where we want to change or improve performance it would be a *minor* release. X.X+1.X.

I would love to add those improvements in a separate PR if that is fine @wrsenn @taylor-knapp 🙏

Reply:
Hmm. I'm not a semver expert but [the docs](https://semver.org/) make it seem that `minor` version changes would indicate the addition of non-breaking functionality.

> MINOR version when you add functionality in a backwards compatible manner
> PATCH version when you make backwards compatible bug fixes

Is it possible that performance flaws could be considered bugs and thus still qualify as a patch release? 🤔 That's basically what we did with the last performance related fix here. (Yes, performance slow enough for large objects it was in essence a bug -- that could be the case here as well depending on the object?)

Reply:
More so, [here's how they define a patch](https://semver.org/#spec-item-6):

> Patch version Z (x.y.Z | x > 0) MUST be incremented if only backwards compatible bug fixes are introduced. A bug fix is defined as an internal change that fixes incorrect behavior.

I think this qualifies as a patch because it is an _internal change that fixes incorrect behavior_.

Reply:
This current PR would be a patch. Changing the behavior of `.reduce` calls in a way that doesn't change the outcome should be as well, but that's a separate changeset.

Reply:
@taylor-knapp @wrsenn  I have tried the suggestion above and those not add any performance improvement for large array scenarios on `returnItems`. 

I have run unit test checking the times for big arrays and they were passing with both versions of the code.

```js
const before = {
            returnInfo: Array(100)
              .fill(null)
              .map((a, index) => ({
                items: Array(50)
                  .fill(null)
                  .map((b, index2) => {
                    return {
                      id: `id-${index}-${index2}`,
                      shipmentState: 'Returned',
                      paymentState: 'Initial',
                    }
                  }),
              })),
          }
          const now = {
            returnInfo: Array(100)
              .fill(null)
              .map((a, index) => ({
                items: Array(50)
                  .fill(null)
                  .map((b, index2) => {
                    return {
                      id: `id-${index}-${index2}`,
                      shipmentState: 'Returned',
                      paymentState: 'Refunded',
                    }
                  }),
              })),
          }

          const start = performance.now()
          orderSync.buildActions(now, before)
          const end = performance.now()

          expect(end - start).toBeLessThan(200)
```
Are we okay with going with the suggestion anyways and ping again Dan to merge/publish this?

Reply:
> (low priority) Using index for an object key/prop confused me when first looking at this.

I think we need to use index, as `key` is already a defined variable above in the code and prettier complains 😅

Reply:
Just to clarify, you're saying you see no performance improvement, but the changes have already been made and you want to proceed with that?